app/views/samples/_upload_results_js.haml

```diff
@@ -56,11 +56,16 @@
                 var lines = text.split(/\r\n|\r|\n/); // tolerate both Windows and Unix linebreaks
                 var samples_param = [] // params for GET call
                 var samples = [] // samples uuids
-
+                var duplicates = false
                 lines.forEach(function(line) { 
                     if (line != '' && isUUID(line.split(',')[0])){
-                        samples_param.push( 'uuids[]=' + line.split(',')[0] );
-                        samples.push( line.split(',')[0] );
+                        if (samples.indexOf(line.split(',')[0]) == -1){
+                            samples_param.push( 'uuids[]=' + line.split(',')[0] );
+                            samples.push( line.split(',')[0] );
```

Comment:
Maybe memoize `var row = line.split(',')` so we don't split & allocate the same array 4 times (starting from line 61)?